src/programs/ectrans-benchmark.F90

```diff
@@ -1394,6 +1450,172 @@ end subroutine dump_gridpoint_field
+  real(kind=jprb), optional :: zspdiv (:,:)
+  real(kind=jprb), optional :: zspscalar (:,:)
+  real(kind=jprb), optional :: zspsc3a (:,:,:)
+  real(kind=jprb), optional :: zspsc2 (:,:)
+  integer(kind=jpib) :: icrc
+  integer(kind=jpim) :: jlev, jfld
+  real(kind=jprb), allocatable :: gfld(:,:)
+  real(kind=jprb), allocatable :: gspfld(:,:)
+  logical :: exist = .false.
+
+  if (myproc == 1) then
+      if (jstep>1)  inquire(file = filename, exist = exist)
+        if (exist) then
+          open(noutdump, file = filename, status="old", position="append", action="write")
+        else
+          open(noutdump, file = filename, action="write")
+      endif
+
+    write(noutdump,*) "===================="
+    write(noutdump,*) "iteration", jstep
+    write(noutdump,*) "===================="
+
+    if (present(zgp) .or. present(zgpuv) .or. present(zgp3a) .or. present(zgp2)) then
+      allocate(gfld(ngptotg,1))
+    endif
+    if (present(zspdiv) .or. present(zspvor) .or. present(zspscalar) &
+       & .or. present(zspsc3a) .or. present(zspsc2)) then
+      allocate(gspfld(1,nspec2g))
+    endif
+  endif
+
+  if (present(zgp)) then
+    icrc = 0
+    do jfld = 1, size (zgp, 2)
+        call gath_grid(pgpg=gfld,kproma=nproma,kfgathg=1,kto=(/1/),KRESOL=1,pgp=zgp(:,jfld:jfld,:))
+        if (myproc == 1) then
+            call crc64 (gfld (:, :), int (size (gfld (:, :)) * kind (gfld), 8), icrc)
+            write (noutdump, '(a," (",i0,") = ",z16.16)') "zgp", jfld, icrc
+        endif
+    enddo
+  endif
+
+  if (present(zgpuv)) then
+    icrc = 0
+    do jfld = 1, size (zgpuv, 3)
+      do jlev = 1, size (zgpuv, 2)
+        call gath_grid(pgpg=gfld,kproma=nproma,kfgathg=1,kto=(/1/),KRESOL=1,pgp=zgpuv(:,jlev:jlev,jfld, :))
+        if (myproc == 1) then
+            call crc64 (gfld (:, :), int (size (gfld (:, :)) * kind (gfld), 8), icrc)
+            write (noutdump, '(a," (",i0,", ",i0,") = ",z16.16)') "zgpuv", jlev, jfld, icrc
+        endif
+      enddo
+    enddo
+  endif
+
+  if (present(zgp3a)) then
+    icrc = 0
+    do jfld = 1, size (zgp3a, 3)
+      do jlev = 1, size (zgp3a, 2)
+        call gath_grid(pgpg=gfld,kproma=nproma,kfgathg=1,kto=(/1/),KRESOL=1,pgp=zgp3a(:,jlev:jlev,jfld, :))
+        if (myproc == 1) then
+            call crc64 (gfld (:, :), int (size (gfld (:, :)) * kind (gfld), 8), icrc)
+            write (noutdump, '(a," (",i0,", ",i0,") = ",z16.16)') "zgp3a", jlev, jfld, icrc
+        endif
+      enddo
+    enddo
+  endif
+
+  if (present(zgp2)) then
+    icrc = 0
+    do jfld = 1, size (zgp2, 2)
+        call gath_grid(pgpg=gfld,kproma=nproma,kfgathg=1,kto=(/1/),KRESOL=1,pgp=zgp2(:,jfld:jfld,:))
+        if (myproc == 1) then
+            call crc64 (gfld (:, :), int (size (gfld (:, :)) * kind (gfld), 8), icrc)
+            write (noutdump, '(a," (",i0,") = ",z16.16)') "zgp2", jfld, icrc
+        endif
+    enddo
+  endif
+
+  if (present(zspvor)) then
+    icrc = 0
+    do jfld = 1, size (zspvor, 1)
```

Comment:
```suggestion
    do jfld = 1, size (ivset, 1)
```
We want to call `gath_spec` on all fields across all tasks, not just the ones on the local task.

Reply:
Fix applied in 8a54ff7